### DISCUSSION TITLE
Marking fields with metadata

### DISCUSSION THREAD

**Discussion by ChristianMay21** [prisma/prisma#17189] [Q&A] [answered]
Howdy everyone - looking for advice.

In my schema, I have an object with multiple fields. I am writing a program that needs to parse some of them, but not all of them. I am hoping to figure out some way to mark which fields should be parsed, and which should not - ideally in the schema file. Ideally, I'd also love to add a bit of metadata that would help the parser determine how exactly they should be parsed.

It looks like it's not possible to create a custom attribute to "mark" which fields should be parsed, but it might be possible to accomplish this with comments, which I believe I can parse using the internal DMMF variable - however, my understanding is that that variable is not intended for external use, so I'd  prefer a way to accomplish this that is more likely to be stable.

 Has anyone tackled a similar problem and has some advice?

**Comment by nurul3101** [comment, marked as answer]
Hello @ChristianMay21 👋 

Just to confirm, you just want specific fields from the object that should not be parsed, but need those fields in the generated Prisma Client? You are not looking for [@ignore](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#ignore) attribute, right?

For adding custom metadata, you should be able to use comments. [zod-prisma](https://github.com/CarterGrimmeisen/zod-prisma) uses comments extensively to pass metadata.

**Comment by ChristianMay21** [comment]
Hi @nurul3101 - saying I wanted to parse some but not all of the fields wasn't strictly correct - all of the fields will be used in different parts of my application, but I want to mark a subset of them to be used in a portion of the application's UI. The @ignore attribute doesn't quite fit my use case, but I didn't know about it and I'm glad I do now! I'm sure it will come in handy in the future.

That's great to know - I do have a simple proof-of-concept for my use-case up-and-running now using comments, but wanted to make sure there wasn't some more "intended" way to add metadata. Now that I know it's how other tools are doing it, I'm much more confident doing it that way 🙂

**Reply by nurul3101** [reply]
I am glad to hear that! 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
